designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsService.scala

```diff
@@ -130,7 +134,9 @@ class UsageStatisticsReportsSettingsService(
       val maybeUrls = for {
         queryParams <- determineQueryParams()
         fingerprint <- new EitherT(fingerprintService.fingerprint(config, nuFingerprintFileName))
-        urls        <- EitherT.pure[Future, StatisticError](statisticsUrls.prepare(fingerprint, queryParams))
+        urls <- EitherT.pure[Future, StatisticError](
+          statisticsUrls.prepare(fingerprint, Instant.now.hashCode.toHexString, queryParams)
```

Comment:
1. I think UUID is better, buy why you choose that, is special reason for it?.
2. I would extract method to calculate correlationId.
3. Did you think what if 2 separate instances will generate URL at the same time? Maybe as a correlationId on elastic we will use fingerprint + correlationId? What you think? (Maybe hashCode would work together with fingerprint but I do not conclusion about using that hashcode)